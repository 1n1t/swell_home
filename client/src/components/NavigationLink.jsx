import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

class NavigationLink extends Component {
	state = {
		color: this.defaultColor
	};

	onMouseEnterHandler = () => {
		this.props.changeOverlayColor(this.props.overlayColor);
		this.setState({ color: this.props.overlayColor });
	};

	onMouseLeaveHandler = () => {
		this.props.changeOverlayColor();
		this.setState({ color: this.defaultColor });
	};

	onClickHandler = () => {
		this.props.onClickCallback();
	};

	defaultColor = '#101010';

	render() {
		const { color } = this.state;
		return (
			<li className="link">
				<NavLink
					exact
					to={this.props.path}
					activeStyle={{ color: this.props.overlayColor }}
					onMouseEnter={this.onMouseEnterHandler}
					onMouseLeave={this.onMouseLeaveHandler}
					onClick={this.onClickHandler}
					data-test="nav-link"
					style={{ color }}
				>
					{this.props.text}
				</NavLink>
			</li>
		);
	}
}

NavigationLink.propTypes = {
	path: PropTypes.string.isRequired,
	text: PropTypes.string.isRequired,
	overlayColor: PropTypes.string.isRequired,
	changeOverlayColor: PropTypes.func.isRequired,
	onClickCallback: PropTypes.func.isRequired
};

export default NavigationLink;
