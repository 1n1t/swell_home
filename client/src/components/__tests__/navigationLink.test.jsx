import NavigationLink from '../NavigationLink';
import { sel, TestHelper } from '../../../../tests/utils';

const defaultProps = {
	changeOverlayColor() {},
	path: '',
	text: '',
	overlayColor: '',
	to: ''
};

const helper = new TestHelper(NavigationLink, defaultProps);

test('the component mounts with defaults', () => {
	const wrapper = helper.mountComponent({}, true);
	expect(wrapper).toMatchSnapshot();
});

test('it renders the given text', () => {
	const text = 'foo';
	const wrapper = helper.mountComponent({ text }, true);
	const innerText = wrapper
		.find(sel('nav-link'))
		.children()
		.text();

	expect(innerText).toBe(text);
});

test('it call parent handler when user hovers over', () => {
	const changeOverlayColor = jest.fn();
	const overlayColor = 'white';
	const wrapper = helper.mountComponent({ changeOverlayColor, overlayColor }, true);
	const el = wrapper.find(sel('nav-link'));

	el.simulate('mouseenter');
	expect(changeOverlayColor).toHaveBeenLastCalledWith(overlayColor);

	el.simulate('mouseleave');
	expect(changeOverlayColor).toHaveBeenLastCalledWith();

	expect(changeOverlayColor).toHaveBeenCalledTimes(2);
});
