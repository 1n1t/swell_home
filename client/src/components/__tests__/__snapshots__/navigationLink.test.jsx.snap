// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component mounts with defaults 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationLink
    changeOverlayColor={[Function]}
    overlayColor=""
    path=""
    text=""
    to=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <NavLink
        activeClassName="active"
        activeStyle={
          Object {
            "color": "",
          }
        }
        ariaCurrent="true"
        data-test="nav-link"
        exact={true}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": undefined,
          }
        }
        to=""
      >
        
      </NavLink>,
      "className": "link",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "activeStyle": Object {
          "color": "",
        },
        "ariaCurrent": "true",
        "children": "",
        "data-test": "nav-link",
        "exact": true,
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "style": Object {
          "color": undefined,
        },
        "to": "",
      },
      "ref": null,
      "rendered": "",
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NavLink
          activeClassName="active"
          activeStyle={
            Object {
              "color": "",
            }
          }
          ariaCurrent="true"
          data-test="nav-link"
          exact={true}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": undefined,
            }
          }
          to=""
        >
          
        </NavLink>,
        "className": "link",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "activeStyle": Object {
            "color": "",
          },
          "ariaCurrent": "true",
          "children": "",
          "data-test": "nav-link",
          "exact": true,
          "onMouseEnter": [Function],
          "onMouseLeave": [Function],
          "style": Object {
            "color": undefined,
          },
          "to": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
